src/controls/viewPicker/ViewPicker.tsx

```diff
@@ -0,0 +1,200 @@
+            options.push({
+              key: view.Id,
+              text: view.Title
+            });
+          }
+        });
+
+        if (this.props.showBlankOption) {
+          // Provide empty option
+          options.unshift({
+            key: EMPTY_VIEW_KEY,
+            text: '',
+          });
+        } else{
+          // Option to unselect the view
+          options.unshift({
+            key: EMPTY_VIEW_KEY,
+            text: EMPTY_VIEW_KEY
+          });
+        }
+
+        this.setSelectedViews();
+        // Update the current component state
+        this.setState({
+          loading: false,
+          results: options
+        });
+        
+    }
+
+    /**
+     * Set the currently selected views(s);
+     */
+    private setSelectedViews(): void {
+      this._selectedView = cloneDeep(this.props.selectedView);
+
+      this.setState({
+        selectedView: this._selectedView,
+      });
+    }
+
+
+  /**
+   * Fires when a view has been selected from the dropdown.
+   * @param option The new selection.
+   * @param index Index of the selection.
+   */
+  private onChange = (event: React.FormEvent<HTMLDivElement>, option: IDropdownOption, index?: number): void => {
+    const { multiSelect, onSelectionChanged } = this.props;
+
+    if (multiSelect) {
+      let selectedViews = this._selectedView ? cloneDeep(this._selectedView) as string[] : [];
+
+      if (option.selected) {
+        selectedViews.push(option.key.toString());
+      }
+      else {
+        selectedViews = selectedViews.filter(view => view !== option.key);
+      }
+      this._selectedView = selectedViews;
+    }
+    else {
+      this._selectedView = option.key.toString();
+    }
+    if (onSelectionChanged) {
+      onSelectionChanged(cloneDeep(this._selectedView));
+    }
+  }
+
+   
+
+    /**
+     * Renders the ViewPicker controls with Office UI Fabric
+     */
+    public render(): JSX.Element {
+        const { loading, results, selectedView } = this.state;
+        const {className, disabled,multiSelect, label, placeholder} = this.props;
+
+        const options : IDropdownOption[] = results.map(v => ({
+            key : v.key,
+            text: v.text
+        }));
+
+
+
+        const dropdownProps: IDropdownProps = {
+            className,
+            options,
+            disabled: disabled,
+            label,
+            placeholder,
+            onChange: this.onChange,
+          };
+        
+        if(multiSelect){
+          dropdownProps.multiSelect = true;
+          dropdownProps.selectedKeys = selectedView as string [];
+        }else {
+          dropdownProps.selectedKey = selectedView as string;
+        }
+        // Renders content
+        return (
+          <div className={styles.viewPicker}>
+            {loading && <Spinner className={styles.spinner} size={SpinnerSize.xSmall}/>}
+            <Dropdown {...dropdownProps} />
```

Comment:
is it shown during loading? if yes - why?

Reply:
Updated in the latest commit as suggested.

Thanks,
Nishkalank